pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -421,21 +422,28 @@ public void remove() {
   private String getGroupKey(int rawKey) {
     // Specialize single group-by column case
     if (_numGroupByExpressions == 1) {
-      return _dictionaries[0].get(rawKey).toString();
+      return getStringFromDictVal(_dictionaries[0].get(rawKey));
     } else {
       int cardinality = _cardinalities[0];
-      StringBuilder groupKeyBuilder = new StringBuilder(_dictionaries[0].get(rawKey % cardinality).toString());
+      StringBuilder groupKeyBuilder = new StringBuilder(getStringFromDictVal(_dictionaries[0].get(rawKey % cardinality)));
       rawKey /= cardinality;
       for (int i = 1; i < _numGroupByExpressions; i++) {
         groupKeyBuilder.append(AggregationGroupByTrimmingService.GROUP_KEY_DELIMITER);
         cardinality = _cardinalities[i];
-        groupKeyBuilder.append(_dictionaries[i].get(rawKey % cardinality));
+        groupKeyBuilder.append(getStringFromDictVal(_dictionaries[i].get(rawKey % cardinality)));
         rawKey /= cardinality;
       }
       return groupKeyBuilder.toString();
     }
   }
 
+  private String getStringFromDictVal(Object value) {
```

Comment:
don't we have getString method in Dictionary?

Reply:
we should restrict the Hex.encodeHexString to ByteDictionary if possible

Reply:
Good point, will override this method.

Reply:
Updated BytesDictionary's getString method and use it in the groupKeyGenerator